src/Psalm/Internal/Type/TypeExpander.php

```diff
@@ -299,6 +302,56 @@ function ($constant_name) use ($const_name_part): bool {
             return $return_type;
         }
 
+        if ($return_type instanceof TPropertiesOf) {
+            if ($return_type->fq_classlike_name === 'self' && $self_class) {
+                $return_type->fq_classlike_name = $self_class;
+            }
+
+            if ($return_type->fq_classlike_name === 'static' && $self_class) {
+                $return_type->fq_classlike_name = is_string($static_class_type) ? $static_class_type : $self_class;
+            }
+
+            if (!$codebase->classExists($return_type->fq_classlike_name)) {
+                return $return_type;
+            }
+
+            // Get and merge all properties from parent classes
+            $class_storage = $codebase->classlike_storage_provider->get($return_type->fq_classlike_name);
+            $properties_types = $class_storage->properties;
```

Comment:
You might want to refine these to just instance properties? That might be closer to what you're looking for, since PHP tends to treat static properties pretty differently.

Reply:
Good catch!